grails-app/services/org/pih/warehouse/product/ProductGroupService.groovy

```diff
@@ -0,0 +1,28 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/
+package org.pih.warehouse.product
+
+class ProductGroupService {
+
+    def grailsApplication
+
+    def addProductToProductGroup(String productGroupId, String productId, boolean isProductFamily) {
+        def g = grailsApplication.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
+        ProductGroup productGroup = ProductGroup.get(productGroupId)
+        Product product = Product.get(productId)
+        if (productGroup && product) {
+            if (product.productFamily) {
+                throw new IllegalArgumentException(g.message(code: "productGroup.productFamily.duplicate.error", default: "This product already has a product family"))
```

Comment:
You will throw this even if the `productGroup` is not `productFamily` type, but product already has the product family assigned.

Reply:
good point, thanks for catching that